subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -0,0 +1,507 @@
+		/* Set state to IN_PROGRES */
+		//execution_state = status;
+		if (execution_state == IN_PROGRESS &&
+		    fota_state == DOWNLOAD_FIRMWARE) {
+			fota_download_start(hostname, file_path);
+		}
+		else if (execution_state == IN_PROGRESS &&
+		    fota_state == APPLY_FIRMWARE) {
+			//clean up and report status
+			fota_state = NONE; //Maybe keep applying firmware as we haven't rebooted
+			update_job_execution(client,
+					     job_id,
+					     SUCCEEDED,
+					     fota_state,
+					     doc_version_number,
+					     "");
+		}
+		else if (execution_state == SUCCEEDED &&
+		    fota_state == APPLY_FIRMWARE) {
+			//callback_emit(AWS_FOTA_EVT_FINISHED);
+		}
+	}
+	return 0;
+
+}
+
+/**
+ * Return 0 for events which should be further processed by the client
+ * Return >= 1 for events which can be skipped/events handled by the
+ * library.
+ */
+int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
+			      const struct mqtt_evt *evt)
+{
+	int err;
+	switch (evt->type) {
+	case MQTT_EVT_CONNACK:
+		if (evt->result != 0) {
+			return 0;
+		}
+
+		err = construct_notify_next_topic(client->client_id.utf8,
+						  notify_next_topic);
+		if (err) {
+			return err;
+		}
+
+		err = aws_jobs_subscribe_expected_topics(client, true);
+		if (err) {
+			return err;
+		}
+
+		err = update_device_shadow_version(client);
+		if (err) {
+			return err;
+		}
+
+		return 0;
+		/* This expects that the application's mqtt handler will handle
+		 * any situations where you could not connect to the MQTT
+		 * broker.
+		 */
+
+	case MQTT_EVT_DISCONNECT:
+		return 0;
+
+	case MQTT_EVT_PUBLISH: {
+		const struct mqtt_publish_param *p = &evt->param.publish;
+		//get_payload
+		err = publish_get_payload(client, p->message.payload.len);
+
+		err = aws_fota_on_publish_evt(client,
+					      p->message.topic.topic.utf8,
+					      p->message.topic.topic.size,
+					      payload_buf);
+
+	} break;
+
+	case MQTT_EVT_PUBACK:
+		if (evt->result != 0) {
+			LOG_ERR("MQTT PUBACK error %d\n", evt->result);
+			return 0;
+		}
+		/* check evt->param.puback.message_id */
+		/* We expect that the client pubacks*/
+		return 0;
+
+	case MQTT_EVT_SUBACK:
+		if (evt->result != 0) {
+			return 0;
+		}
+
+		/* check evt->param.suback.message_id against SUB_BASE */
+		return 0;
+
+	default:
+		/* Handling for default case? */
+		return 0;
+	}
+	return 0;
+}
+
+void http_fota_handler(enum fota_download_evt_id evt)
+{
+	__ASSERT_NO_MSG(c != NULL);
+
+	switch(evt) {
+		case FOTA_DOWNLOAD_EVT_FINISHED:
+			fota_state = APPLY_FIRMWARE;
+			update_job_execution(c,
+					     job_id,
+					     IN_PROGRESS,
+					     fota_state,
+					     doc_version_number,
+					     "");
+
+			break;
+		case FOTA_DOWNLOAD_EVT_ERROR:
+			update_job_execution(c,
+					     job_id,
+					     FAILED,
+					     fota_state,
+					     doc_version_number,
+					     "");
+			//Emit AWS_FOTA_ERR
+			break;
+	}
+
+}
+
+
+/**@brief Initialize the AWS Firmware Over the Air library.
```

Comment:
Docs to the API functions should be present in the header file.

Reply:
It will be eventually :)

Reply:
Added some docs to the header.